### PR TITLE
Add X-Ray Engine

### PR DESCRIPTION
**Pull request by diedyesterdaywashere**
### SteamDB app page links to a few games using this

https://steamdb.info/franchise/S.T.A.L.K.E.R./

### Brief explanation of the change

Added a rule for X-Ray, engine that powered S.T.A.L.K.E.R

### CODE REVIEW

**Comment by diedyesterdaywashere** [comment]
@Lifeismana could you pls help, i can't figure out whats the issue with it

**Comment by diedyesterdaywashere** [comment]
nvm im just very stupid

File: tests/types/Engine.X-Ray.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+xrGame.dll
```
Comment (Lifeismana) [inline tests/types/Engine.X-Ray.txt:1]: you should add a test case of that file under a folder 
e.g: https://steamdb.info/depot/4501/

**Review by Lifeismana — CHANGES_REQUESTED** [review]

File: tests/types/Engine.X-Ray.txt (hunk 1 above)
Comment (diedyesterdaywashere) [reply tests/types/Engine.X-Ray.txt:1]: oh ye forgot about it

**Comment by diedyesterdaywashere** [comment]
done

**Review by Lifeismana — APPROVED** [review]

**Comment by Lifeismana** [comment]
Looks good to me now, thanks for the PR!

**Pull request merged**